be/src/exec/vectorized/sorting/sort_column.cpp

```diff
@@ -98,6 +112,230 @@ class ColumnSorter final : public ColumnVisitorAdapter<ColumnSorter> {
     bool _build_tie;
 };
 
+// Sort multiple a column from multiple chunks(vertical column)
+class VerticalColumnSorter final : public ColumnVisitorAdapter<VerticalColumnSorter> {
+public:
+    explicit VerticalColumnSorter(const bool& cancel, const std::vector<ColumnPtr>& columns, bool is_asc_order,
+                                  bool is_null_first, Permutation& permutation, Tie& tie, std::pair<int, int> range,
+                                  bool build_tie, size_t limit)
+            : ColumnVisitorAdapter(this),
+              _cancel(cancel),
+              _is_asc_order(is_asc_order),
+              _is_null_first(is_null_first),
+              _vertical_columns(columns),
+              _permutation(permutation),
+              _tie(tie),
+              _range(range),
+              _build_tie(build_tie),
+              _limit(limit),
+              _pruned_limit(permutation.size()) {}
+
+    size_t get_limited() const { return _pruned_limit; }
+
+    Status do_visit(const vectorized::NullableColumn& column) {
+        std::vector<NullData> null_datas;
+        std::vector<ColumnPtr> data_columns;
+        for (auto& col : _vertical_columns) {
+            auto real = down_cast<const NullableColumn*>(col.get());
+            null_datas.push_back(real->immutable_null_column_data());
```

Comment:
This will copy the whole `NullData`s, each of which is a `std::vector<uint8_t>`. How about 
```C++
std::vector<uint8_t*> null_datas;
null_datas.emplace_back(real->immutable_null_column_data().data());
```

Or 
```C++
std::vector<const NullData*> null_datas;
null_datas.emplace_back(&real->immutable_null_column_data());
```

Reply:
Good idea!